media/js/base/mozilla-utils.js

```diff
@@ -21,6 +21,63 @@ if (typeof window.Mozilla === 'undefined') {
         }
     };
 
+    /**
+     * Get GA data attribute values for download_firefox_thanks() buttons.
+     * @param {Object} window.site
+     * @returns {Object} os, name, version
+     */
+    Utils.getDownloadAttributionValues = function(site) {
+        var data = {};
+        var platform = site.platform;
+
+        switch(platform) {
+        case 'windows':
+            data.os = 'Desktop';
+            data.name = site.isARM ? 'Windows ARM64/AArch64' : 'Windows 32-bit';
+            data.version = site.isARM ? 'win64-aarch64': 'win';
+            break;
+        case 'osx':
+            data.os = 'Desktop';
+            data.name = 'macOS';
+            data.version = 'osx';
+            break;
+        case 'linux':
+            data.os = 'Desktop';
+            data.name = site.archSize === 64 ? 'Linux 64-bit' : 'Linux 32-bit';
+            data.version = site.archSize === 64 ? 'linux64': 'linux';
+            break;
+        case 'ios':
+            data.os = 'iOS';
+            data.name = 'iOS';
+            data.version = 'ios';
+            break;
+        case 'android':
+            data.os = 'Android';
+            data.name = 'Android';
+            data.version = 'android';
+            break;
+        }
+
+        return data;
+    };
+
+    /**
+     * Set platfrom specific GA data attributes for download_firefox_thanks() buttons.
+     */
+    Utils.trackDownloadThanksButton = function() {
+        var downloadButton = document.querySelectorAll('.c-button-download-thanks > a');
+        var data = Utils.getDownloadAttributionValues(window.site);
+
+        for (var i = 0; i < downloadButton.length; ++i) {
+
+            if (data && data.os && data.name && data.version) {
+                downloadButton[i].setAttribute('data-display-os', data.os);
+                downloadButton[i].setAttribute('data-display-name', data.name);
+                downloadButton[i].setAttribute('data-download-version', data.version);
```

Comment:
There isn't a 1:1 mapping for the version defined by the old button to the version defined in the JS here. (For example win64, win-msi, and standard will be reported as something else) Is that okay?

Reply:
Win64 downloads have never been enabled via the dl button (since the stub installer handles detection), and I don't think we'd ever serve msi installers via /thanks. So I think this is OK.

Reply:
There's not a lot of them but win64, win64-msi are currently being [reported in GA](https://analytics.google.com/analytics/web/#/report/content-event-events/a36116321w64087799p65789850/explorer-table.secSegmentId=analytics.customDimension27&explorer-table.plotKeys=%5B%5D&explorer-table.rowCount=250&explorer-table.filter=win). I don't see standard though, even though it is documented in the read the docs.

Reply:
There’s a modal on /windows that enables downloading specific builds. On /browsers/windows-64-bit/ my guess is they are following the link to the /all page.

I don’t think any of these events are coming from download buttons since those links have never been enabled to appear.

Reply:
If you[ filter to /firefox/new](https://analytics.google.com/analytics/web/#/report/content-event-events/a36116321w64087799p65789850/explorer-table.plotKeys=%5B%5D&explorer-table.secSegmentId=analytics.customDimension27&_r.drilldown=analytics.eventCategory:~2Ffirefox~2Fnew~2F%20Interactions,analytics.eventAction:Firefox%20Download/) you can see they are being triggered on /new 🤷‍♀️

Reply:
Is that the modal there too?

Reply:
Hm, also seeing them on [/firefox](https://analytics.google.com/analytics/web/#/report/content-event-events/a36116321w64087799p65789850/_r.drilldown=analytics.eventCategory:~2Ffirefox~2F%20Interactions&explorer-table.secSegmentId=analytics.customDimension27&explorer-table.plotKeys=%5B%5D/). No modal there.

Reply:
The older version of /new has the modal yes. Are these session based events? That may explain things. Still think it this is likely a red herring.